### PR TITLE
  Crossref Api and formatting changes

### PR DESCRIPTION
**Pull request by JoeAmmar**
Added author name formatting to DOILookup (with import in ArticleEditor)
Added ?mailto to api calls in crossref to ArticleEditor
Added citation count extraction to ArticleEditor
Added logic for extracting year from crossref based on whether it's a published study or not to to ArticleEditor
Added extraction of preprint_url from crossref

### CODE REVIEW

File: src/components/curateform/DOILookup.jsx [hunk 1]
```diff
@@ -144,4 +144,111 @@ DOILookup.defaultProps = {
...
 };
 
+
+
+
+
+
+export function retrieve_authors(author) {
+
+	Array.prototype.last = function() {
```
[the remark this answers is not in the document]
Comment (JoeAmmar) [reply src/components/curateform/DOILookup.jsx]: Thank you, I'll update using the last function from lodash

File: src/components/curateform/DOILookup.jsx [hunk 2]
```diff
@@ -144,4 +144,111 @@ DOILookup.defaultProps = {
...
+
+	String.prototype.last = function() {
+	 return this[this.length - 1];
+	}
+
+
+
+
+
+  const authors = author;
```
[the remark this answers is not in the document]
Comment (JoeAmmar) [reply src/components/curateform/DOILookup.jsx]: Thanks, nice catch!

File: src/components/curateform/DOILookup.jsx [hunk 3]
```diff
@@ -144,4 +144,111 @@ DOILookup.defaultProps = {
...
+          for (var i = 0; i < authorsLen; i++) {
+            let authConcat = authors[i]['given'] + " " + authors[i]['family']
+            authorList.push(authConcat)
+          };
+           var fullCitation = ''
+
+//Formatting for author loop
+				var beginning = ""
+				 	var end = ""
+					  function authorFormatting(auth, fullCitation, beginning, end){
```
[the remark this answers is not in the document]
Comment (JoeAmmar) [reply src/components/curateform/DOILookup.jsx]: If I'm not mistaken, it's slightly different for 2 vs more than 2 (unless y'all want it formatted another way)

1 auth: A1
2 auth: A1 & A2 (no commas)
3 or more: A1, A2, & A3
7 or more: A1, A2, A3, A4, A5, A6, … , & Alast

I redid all the code following your suggestions. In the process, I took out this function entirely.

File: src/components/curateform/DOILookup.jsx [hunk 4]
```diff
@@ -144,4 +144,111 @@ DOILookup.defaultProps = {
...
+	}
+
+
+
+
+
+  const authors = author;
+  // retrieve list of authors
+           const authorsLen = authors.length;
+          let authorList = []
```
[the remark this answers is not in the document]
Comment (JoeAmmar) [reply src/components/curateform/DOILookup.jsx]: Thanks for the heads up. I formatted it. Let me know if you'd like it modified once I push the update.

File: src/components/curateform/DOILookup.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (eplebel) [reply src/components/curateform/DOILookup.jsx]: >  it's slightly different for 2 vs more than 2 (unless y'all want it formatted another way)

1 auth: A1
2 auth: A1 & A2 (no commas)
3 or more: A1, A2, & A3
7 or more: A1, A2, A3, A4, A5, A6, … , & Alast

yes this is correct.

File: src/components/curateform/DOILookup.jsx [hunk 5]
```diff
@@ -144,4 +144,111 @@ DOILookup.defaultProps = {
...
+          for (var i = 0; i < authorsLen; i++) {
+            let authConcat = authors[i]['given'] + " " + authors[i]['family']
```
[the remark this answers is not in the document]
Comment (JoeAmmar) [reply src/components/curateform/DOILookup.jsx]: Thanks for the heads up! I added this in.

File: src/components/curateform/DOILookup.jsx [hunk 6]
```diff
@@ -144,4 +144,111 @@ DOILookup.defaultProps = {
...
+					 			 end
+					  }
+
+					  return fullCitation
+
+					};
+
+					// Loop to get author names prettied up
+           for (var i = 0; i < authorList.length; i++) {
+             let aut = authorList[i]
```
[the remark this answers is not in the document]
Comment (JoeAmmar) [reply src/components/curateform/DOILookup.jsx]: I completely remade the function (if you'd like it changed again, please let me know).

**Pull request merged**